db/sql/0_msar.sql

```diff
@@ -902,29 +1032,49 @@ All fields are optional, and a null value as input returns 'text'
 */
 SELECT COALESCE(
   -- First choice is the type specified by numeric IDs, since they're most reliable.
-  format_type(typ.id, typ.modifier),
+  format_type((typ_jsonb ->> 'id')::integer, (typ_jsonb ->> 'modifier')::integer),
   -- Second choice is the type specified by string IDs.
-  COALESCE(
-    msar.get_fully_qualified_object_name(typ.schema, typ.name),
-    typ.name,
-    'text'  -- We fall back to 'text' when input is null or empty.
-  )::regtype::text || COALESCE(  -- This section builds the type options blob.
-    '(' || topts.length || ')',
-    ' ' || topts.fields || ' (' || topts.precision || ')',
-    ' ' || topts.fields,
-    '(' || topts.precision || ', ' || topts.scale || ')',
-    '(' || topts.precision || ')',
-    ''
-  ) || CASE WHEN topts.array_ THEN '[]' ELSE '' END
+  __msar.get_formatted_base_type(
+    COALESCE(
+      msar.get_fully_qualified_object_name(typ_jsonb ->> 'schema', typ_jsonb ->> 'name'),
+      typ_jsonb ->> 'name',
+      'text'  -- We fall back to 'text' when input is null or empty.
+    ),
+    typ_jsonb -> 'options'
+  ) || CASE WHEN (typ_jsonb -> 'options' ->> 'array')::boolean THEN '[]' ELSE '' END
```

Comment:
This function seems shorter, which is nice, but it's become more horizontally intense. Could you split some of the lines? Examples of horizontally intense expressions: `) || CASE WHEN ... THEN ...`, `format_type((typ_jsonb ->> 'id')::integer, (typ_jsonb ->> 'modifier')::integer)`.

Reply:
Okay, I split a couple of lines up a bit. I did leave one of the function calls on a single line, because splitting it up made it harder (for me) to see the nesting in that case.